internal/handlers/auth.go

```diff
@@ -63,6 +65,69 @@ var (
 	}
 )
 
+func (a *AuthHandler) Authenticate(c *fiber.Ctx) error {
+	var input helpers.AuthenticateUser
+	if err := c.BodyParser(&input); err != nil {
+		return helpers.Dispatch500Error(c, err)
+	}
+
+	user, userExist, err := a.userRepository.FindUserByCondition("email", input.Email)
+	if err != nil {
+		return helpers.Dispatch500Error(c, err)
+	}
+	if !userExist {
+		return helpers.Dispatch400Error(c, "invalid account credentials", nil)
+	}
+
+	hashedPassword := []byte(user.Password)
+	plainPassword := []byte(input.Password)
+	err = bcrypt.CompareHashAndPassword(hashedPassword, plainPassword)
+
+	if err != nil {
+		return helpers.Dispatch400Error(c, "email and password does not match", nil)
+	}
+	jwtToken, err := helpers.GenerateToken(constant.JWTSecretKey, user.Email, user.FirstName+" "+user.LastName)
+
+	if err != nil {
+		return helpers.Dispatch400Error(c, "something went wrong", err)
+	}
+
+	// check if account is active
+	if user.AccountStatus == int(Suspended) {
```

Comment:
@CeoFred Just wondering why the `AccountStatus` struct isn't of type `AccountStatus` for direct comparison as opposed to `int` wrapping...direct comparison enforces consistency.

Reply:
`user.AccountStatus` comes back from the db as `int` and not `AccountStatus` type which is still an int type though. @opensaucerer

Reply:
@CeoFred Yeah...however, if the type specified on the struct is `AccountStatus` which is mapped against an `int` type as I think it currently is, then the db formatted should format it against `AccountStatus` without any issue...else, one could also intercept the formatter for `AccountStatus` quite easily. I think gorm exposes a serializer interface for this....but ideally, it shouldn't be needed as the type `AccountStatus` here is a simple type mapped against the `int` primitve, and `gorm` should just be able to serialize into it easily.